core/src/main/java/org/mule/runtime/core/util/IOUtils.java

```diff
@@ -262,4 +272,60 @@ public static long copyLarge(Reader input, Writer output) throws IOException
         }
         return count;
     }
+
+    /**
+     * Transforms an Object attachment into a DataHandler of its corresponding type.
+     *
+     * @param name        the name of the attachment being handled
+     * @param object      the attachment to be handled
+     * @param contentType the Content-Type of the attachment that is being handled
+     * @return a {@link DataHandler} of the corresponding attachment
+     * @throws Exception if the transformation fails.
+     */
+    public static DataHandler toDataHandler(String name, Object object, String contentType) throws Exception
```

Comment:
I see two methods here, one without `contentType` and another one that overrides it which receives a `contentType`and fails if it is null. Since almost every branch has a different implementation for null contentType, we should not even delegate one method to the other passing `null` to the content, but instead group all those branches an simplify this method.